numba_dpex/spirv_generator.py

```diff
@@ -86,22 +86,12 @@ def generate(self, llvm_spirv_args, ipath, opath):
     def _llvm_spirv():
         """Return path to llvm-spirv executable."""
         result = None
+        try:
+            import llvm_spirv
+        except ImportError:
+            raise ImportError("Cannot import llvm-spirv")
```

Comment:
A more descriptive error message is needed here. The message implies that it is something wrong with user env, may be they need to install llvm-spirv. 

We should be clear that it is an internal numba-dpex error.

Reply:
I am adding a new `Exception` to dpex_exceptions called `UnreachableError`. The idea is borrowed from LLVM, it indicates that a supposedly impossible branch inside the compiler got executed. 

We can leave the code here as is and change it to an `UnreachableError` later. The idea being that this is an exception that should never be thrown.

Reply:
@mingjie-intel refer #798